packages/google_sign_in/lib/src/device_flow_widget.dart

```diff
@@ -0,0 +1,127 @@
+// Copyright 2022 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+
+/// The [GlobalKey] that identifies a [NavigatorState].
+///
+/// This object refers to a navigator used to push the "Device Flow Widget"
+/// that shows "verification url" and "user code" which are
+/// required to authorize sign-in.
+///
+/// The navigatorKey.currentContext is non-null if navigatorKey
+/// is assigned to a valid [Navigator].
+GlobalKey<NavigatorState> navigatorKey = GlobalKey<NavigatorState>();
+
+/// Closes the widget that was shown from [showDeviceFlowWidget].
+void closeDeviceFlowWidget() {
+  navigatorKey.currentState!.pop();
+}
+
+/// Displays a widget that shows [code] and [verificationUrl].
+void showDeviceFlowWidget({
+  required String code,
+  required Uri verificationUrl,
+  required Duration expiresIn,
+  VoidCallback? onExpired,
+  VoidCallback? onCanceled,
+}) {
+  showDialog<void>(
+    context: navigatorKey.currentContext!,
+    barrierDismissible: false,
+    builder: (BuildContext context) {
+      return AlertDialog(
+        title: const Text('Google SignIn'),
```

Comment:
The UI doesn't fit on a watch screen.

<img src="https://user-images.githubusercontent.com/40190588/182741500-0651a41b-2c88-4802-b192-07438064bd3f.png" width="240">

Also
```suggestion
        title: const Text('Google Sign In'),
```

Reply:
Responsive layout for different sized screens and a better design is in progress on a different branch: https://github.com/HakkyuKim/plugins/commit/2bc826793a15d9e581f3df6b375d11d686072c24. I've planned it as a separate PR.

![Screenshot from 2022-08-04 11-07-46](https://user-images.githubusercontent.com/43136596/182747631-26320d59-3e32-4b45-8571-04f0bd7fdabf.png)